src/libs/Tokenizer.js

```diff
@@ -0,0 +1,95 @@
+
+
+var theTokenizer;
+
+class Tokenizer {
+
+    constructor(program, literalsList) {
+        this.program = program;
+        this.literals = literalsList;
+        this.tokens = [];
+        this.currentToken = 0;
+        this.tokenize();
+    }
+
+    tokenize() {
+        let tokenizedProgram = this.program;
+        tokenizedProgram = tokenizedProgram.replace(/\n/g, "");
+
+        // extracts replacing them with _$_, stores them in an array
+        // to be added back in later.
+        var texts = [...tokenizedProgram.matchAll(/".*"/g)];
+        tokenizedProgram = tokenizedProgram.replace(/".*"/g, "_$_");
+
+        // prevents leading 'lat' in lat lon from combining with variable name token
+        tokenizedProgram = tokenizedProgram.replace(/( \d+)/g, function(m){
+            return "_" + m;
+        });
+        console.log(tokenizedProgram);
+
+        this.literals.forEach((s) => {
+            let regex = new RegExp(s, "g");
+            tokenizedProgram = tokenizedProgram.replace(regex, "_"+s+"_");
+            console.log(tokenizedProgram);
+        });
+        //TODO: Below will make lat and lon hard to distinguish?? may need comma between them??
+        tokenizedProgram = tokenizedProgram.replace(/ /g, "");
+
+        texts.forEach((text) => {
+            tokenizedProgram = tokenizedProgram.replace(/\$/, text);
+        });
+
+        tokenizedProgram = tokenizedProgram.replace(/__/g, "_");
```

Comment:
how about we do `/[_]+/g` for the regex?

Reply:
Good idea! much more robust.

Reply:
Thanks for updating the code for this!